src/deserialize.zig

```diff
@@ -103,19 +124,23 @@ pub fn deserialize(comptime T: type, serialized: []const u8, out: *T) !usize {
         },
         .Pointer => |ptr| switch (ptr.size) {
             .Slice => if (ptr.child == u8) {
-                if (serialized[0] < rlpByteListShortHeader) {
-                    out.* = serialized[0..1];
-                    return 1;
-                } else if (serialized[0] < rlpByteListLongHeader) {
-                    const size = @as(usize, serialized[0] - rlpByteListShortHeader);
-                    out.* = serialized[1 .. 1 + size];
-                    return 1 + size;
-                } else {
-                    const size_size = @as(usize, serialized[0] - rlpByteListLongHeader);
-                    const size = readIntSliceBig(usize, serialized[1 .. 1 + size_size]);
-                    out.* = serialized[1 + size_size .. 1 + size_size + size];
-                    return 1 + size + size_size;
-                }
+                // if (serialized[0] < rlpByteListShortHeader) {
+                //     out.* = serialized[0..1];
+                //     return 1;
+                // } else if (serialized[0] < rlpByteListLongHeader) {
+                //     const size = @as(usize, serialized[0] - rlpByteListShortHeader);
+                //     out.* = serialized[1 .. 1 + size];
+                //     return 1 + size;
+                // } else {
+                //     const size_size = @as(usize, serialized[0] - rlpByteListLongHeader);
+                //     const size = readIntSliceBig(usize, serialized[1 .. 1 + size_size]);
+                //     out.* = serialized[1 + size_size .. 1 + size_size + size];
+                //     return 1 + size + size_size;
+                // }
+                var r = sizeAndDataOffset(serialized);
+                std.debug.print("{} {}\n", .{ r.offset, r.size });
+                out.* = serialized[r.offset .. r.offset + r.size];
```

Comment:
[Cross-linking comment.](https://github.com/gballet/zig-rlp/pull/11#issuecomment-1849935568)

Reply:
yup that makes sense. I'll do it in another PR though.